src/Feed/WalkerProgress.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Walker Progress class.
+ *
+ * @package Automattic\WooCommerce\ProductFeedForOpenAI
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\ProductFeedForOpenAI\Feed;
+
+/**
+ * Simple class that tracks/indicates the progress of a walker.
+ */
+class WalkerProgress {
+	/**
+	 * Total number of items to process.
+	 *
+	 * @var int
+	 */
+	public int $total_count;
+
+	/**
+	 * Total number of batches to process.
+	 *
+	 * @var int
+	 */
+	public int $total_batch_count;
+
+	/**
+	 * Number of items processed so far.
+	 *
+	 * @var int
+	 */
+	public int $processed_items = 0;
+
+	/**
+	 * Number of batches processed so far.
+	 *
+	 * @var int
+	 */
+	public int $processed_batches = 0;
+
+	/**
+	 * Creates a WalkerProgress instance from a WooCommerce products query result.
+	 *
+	 * @param object $result The result object from wc_get_products().
+	 * @return static
+	 */
+	public static function from_wc_get_products_result( object $result ): static {
```

Comment:
`static` is only avaialbe since PHP 8.0 :D 

```suggestion
	public static function from_wc_get_products_result( object $result ): WalkerProgress {
```

Reply:
Why is PHPCS not highlighting this?! 😿 

Fixed in https://github.com/woocommerce/OpenAI-Product-Feed/pull/13/commits/28a698d497b7d353b91e50ad24175e3df2a7b0a5